### ISSUE TITLE
Support .net core

### ISSUE THREAD

**Issue by feliwir** [Ruslan-B/FFmpeg.AutoGen#39]
Hey, it would be nice if it was also possible to generate Bindings for .net core

**Comment by Ruslan-B** [comment]
I'll take a look, just wondering myself does the PCL infrastructure supporting interop and usafe c#.

**Comment by feliwir** [comment]
I'd suggest using a .net standard library. It runs on .Net core, Xamarin and .Net framework

**Comment by Ruslan-B** [comment]
after short investigation looks like  we need .NET Standard 2.0 in order to get ICustomMarshaler for ConstCharPtrMarshaler. And this working only for Visual Studio 2017. Host stuff! :)

**Comment by Ruslan-B** [comment]
bah just figured out - @feliwir I beg a pardon but do you have native ffmpeg for any of Xamarin platforms?

**Comment by feliwir** [comment]
No sorry :( I am not doing any mobile programming normally

**Comment by Ruslan-B** [comment]
But then why? do you need it for core

**Comment by feliwir** [comment]
No, i just said that .net standard bindining works on all those platforms.

**Comment by feliwir** [comment]
Yes, i need it for .net core

**Comment by Ruslan-B** [comment]
Right now you can reassemble the project yourself it should work, Meanwhile I'll wait for official release of VS2017 and after I can consider to update to standard 2.0.

**Comment by Ruslan-B** [comment]
It will be possible only after .NET Core 2.0 release, as ICustomMarshaler is not supported in .NET Core 1.1

**Comment by Ruslan-B** [comment]
I added experimental support.

**Comment by Ruslan-B** [comment]
Install-Package FFmpeg.AutoGen -Version 3.3.3.1

**Comment by CraigN** [comment]
I see you targeted .NET Core 2.0, would it not be better to target .NET Standard 2.0 instead? See [here](https://github.com/dotnet/standard/blob/master/README.md)

**Comment by Ruslan-B** [comment]
Microsoft really confuses me some times, fair point i'll retarget to netstandard2.0 instead.

**Comment by Ruslan-B** [comment]
Or even better I'll add another target :).

**Comment by CraigN** [comment]
You're not the only one. The nice thing about the .NET Standard 2.0 is that you can publish against it and consume from .NET Core 2.0, .NET Framework 4.6.1+, etc. Think of it more like an API version.

**Comment by Ruslan-B** [comment]
From now binding are netstandard2.0 only.

**Comment by popcatalin81** [comment]
@yoav85 Please feel free to open a new issue with appropriate description and reproduction steps if you think the problems you're encountering are because of FFMpeg.Autogen. 

This issue was closed and it's not a good place for discussion.
